src/main/java/net/sf/jabref/logic/util/version/VersionInformation.java

```diff
@@ -0,0 +1,184 @@
+                    checkForUpdatesAutomatically();
+                }
+                break;
+            case WEEKLY:
+                if (lastCheckedDate.plusWeeks(1).isBefore(currentDate)) {
+                    checkForUpdatesAutomatically();
+                }
+                break;
+            case MONTHLY:
+                if (lastCheckedDate.plusMonths(1).isBefore(currentDate)) {
+                    checkForUpdatesAutomatically();
+                }
+                break;
+            case NEVER:
+                break;
+        }
+    }
+
+    /**
+     * Checks if a newer version is available
+     *
+     * @param forceCheck      when true check the updateserver again even if it already has been in this instance
+     * @param respectIgnore   when true the ignored Update set by the user will be respected
+     * @param manualExecution when true there will be a popup if something goes wrong
+     * @return if there is a newer versin available
+     */
+    public boolean isNewerVersionAvailable(boolean forceCheck, boolean respectIgnore, boolean manualExecution) {
+        // if the ignored update is not newer than the current version remove the ignored version
+        //this will be only neede after updating JabRef
+        Version ignoredVersion = new Version(Globals.prefs.get(JabRefPreferences.VERSION_IGNORED_UPDATE));
+        if (!ignoredVersion.isNewerThan(Globals.BUILD_INFO.getVersion())){
+            Globals.prefs.put(JabRefPreferences.VERSION_IGNORED_UPDATE, "");
+        }
+
+        if (forceCheck || latestVersion == null) {
+            grabLatestVersion(manualExecution);
+        }
+
+        // if we couldn't grab the latest version
+        if (latestVersion == null){
+            return false;
+        }
+
+        // if this update has already been ignored by the user ignore it again, or don't
+        if (respectIgnore && latestVersion.getFullVersion().equals(Globals.prefs.get(JabRefPreferences.VERSION_IGNORED_UPDATE))) {
+            return false;
+        }
+        return latestVersion.isNewerThan(Globals.BUILD_INFO.getVersion());
+    }
+
+    /**
+     * Grabs the latest version of JabRef from GitHub
+     * the tag name on GitHub has to be in form of: 'vX.X' or 'X.X' where X is any number
+     * this version needs some time so it could halt the application if it's not run on its own
+     *
+     * @param manualExecution if true a Dialog will popup if the update check fails
+     */
+    private void grabLatestVersion(boolean manualExecution) {
+        try {
+            URLConnection connection = new URL(JABREF_GITHUB_URL).openConnection();
+            connection.setRequestProperty("Accept-Charset", "UTF-8");
+            BufferedReader rd = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+            JSONObject obj = new JSONObject(rd.readLine());
+            latestVersion = new Version(obj.getString("tag_name").replaceFirst("v", ""));
```

Comment:
I would extract a class which simply gets the latest version, e.g., a VersionChecker, with a method Optional<Version> getLatestVersion() as the request can fail returning no version at all.